src/components/shared/buttons/AddButton.tsx

```diff
@@ -12,9 +14,16 @@ function AddButton() {
 
   return (
     <>
-      <div
-        className="bg-orange flex flex-row justify-center items-center rounded-lg h-[3.5rem] w-full shadow-[0_1.5rem_1rem] shadow-pale-orange cursor-pointer"
+      <button
+        className={` flex flex-row justify-center items-center rounded-lg h-[3.5rem] w-full shadow-[0_1.5rem_1rem] shadow-pale-orange cursor-pointer  ${
+          isPressed ? "bg-dark-orange" : "bg-orange hover:opacity-80"
+        } `}
         onClick={() => handleClick()}
+        onMouseDown={() => setIsPressed(true)}
+        onMouseUp={() => setIsPressed(false)}
+        onMouseLeave={() => setIsPressed(false)}
+        onTouchStart={() => setIsPressed(true)}
+        onTouchEnd={() => setIsPressed(false)}
```

Comment:
อันนี้ถูกต้องเชิงพฤติกรรมแน่นอน

ปกติ animation ของ UI ถ้ามันไม่ได้ซับซ้อนเขาจะไม่อยากใช้ javascript เข้าช่วยเท่าไหร่ เพราะจะเป็นการใช้ทรัพยากรย์ที่สิ้นเปลือง (เปลือง ram ที่ใช้เก็บ event listener + state)
ทีนี้เรามาดูกันว่าปกติคนอื่นๆเขาจะเขียนยังไง

เราสามารถใช้ css ทำ animation ปุ่มได้ง่ายๆ โดยใช้พวก pseudo-class เช่น :hover, :active, :focus , and so on.
ซึ่งใน Tailwind เราจะใช้เป็น prefix แบบนี้ยกตัวอย่าง 

[Tailwind: Handling Hover, Focus, and Other States](https://tailwindcss.com/docs/hover-focus-and-other-states)

```tsx
<button
   className=" active:bg-dark-orange hover:bg-orange/80 "
>
```

แบบนี้เวลาปุ่มถูกกดก็จะแสดง bg-dark-orange ส่วนเวลา hover ก็จะแสดง bg-orange ที่มี opacity 80
จะเห็นว่าเราไม่ต้องเก็บ/จัดการ state เลย ไม่ต้องใช้ js